pkg/parser/parser.go

```diff
@@ -0,0 +1,121 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/tjgurwara99/monkey-go/pkg/ast"
+	"github.com/tjgurwara99/monkey-go/pkg/token"
+)
+
+type parser struct {
+	tokens chan token.Token
+
+	current token.Token
+	peek    token.Token
+
+	errors []error
```

Comment:
Are you thinking of making it so that the parser is permissive of errors? I think Go's parser does it like this but not sure.

Reply:
This implementation is the one from the book, as I am following it.